tests/test-desktop-files.lua

```diff
@@ -0,0 +1,34 @@
+local runner = require("_runner")
+local menubar = require("menubar")
+local gears_debug = require("gears.debug")
+
+local assets_dir = os.getenv("TEST_ASSETS") .. "/"
+menubar.menu_gen.all_menu_dirs = { assets_dir }
+
+local menu_gen_list = nil
+
+runner.run_steps({
+    function()
+        menubar.menu_gen.generate(function(menu_entries)
+            menu_gen_list = menu_entries
+        end)
+        return true
+    end,
+    function(count)
+        if menu_gen_list == nil then
+            -- Continue waiting
+            if count >= 5 then
+                return false
+            end
+            return
+        else
+            -- Begin the actual test
+            local active_entry = menu_gen_list[1] -- Lua array counting starts at 1, right?
+            assert(active_entry.icon == "/usr/share/icons/hicolor/128x128/apps/firefox.png")
```

Comment:
What's the first entry of this table supposed to be? I don't see such a .desktop file being added anywhere in this commit. Does the test only work if Firefox is installed? Well, Firefox is not installed on Travis.